samcli/lib/sync/flows/alias_version_sync_flow.py

```diff
@@ -90,13 +93,18 @@ def compare_remote(self) -> bool:
 
     def sync(self) -> None:
         function_physical_id = self.get_physical_id(self._function_identifier)
+        current_alias_version = self._get_version_alias_if_exists()
         version = self._lambda_client.publish_version(FunctionName=function_physical_id).get("Version")
         self._local_sha = str_checksum(str(version), hashlib.sha256())
         LOG.debug("%sCreated new function version: %s", self.log_prefix, version)
         if version:
             self._lambda_client.update_alias(
                 FunctionName=function_physical_id, Name=self._alias_name, FunctionVersion=version
             )
+            if self._delete_old_alias and current_alias_version:
+                function_name_w_version = "{}:{}".format(function_physical_id, current_alias_version)
```

Comment:
nit: changing `_w_` to `_with_`, or renaming to something like `formatted_function_name_version` for future contributors that are not familiar with the "with" shorthand